packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -464,11 +469,15 @@ export function initialize(
     const stopHandlingPointerMoveEvent = dispatch(startHandlingPointerMoveEvent())
     const stopPollingElementFromPoint = dispatch(startPollingElementFromPoint())
     const unregisterBuilderComponents = dispatch(registerBuilderComponents())
-
     const breakpoints = ReactPage.getBreakpoints(getState())
     dispatch(setBreakpoints(breakpoints))
     dispatch(setIsInBuilder(true))
+
+    // This will dispatch any buffered actions, but we don't control the order of the messages.
     channel.dispatchBuffered()
+    channel.initialize()
+    // This will dispatch any buffered actions that need to be run after the builder receive all the previous messages.
+    channel.dispatchBuffered({ lifecycle: Lifecycle.AFTER_INITIALIZED })
```

Comment:
I'd do `channel.initialize()` here and move `dispatchBuffered()` and `dispatchBuffered({ lifecycle: Lifecycle.AFTER_INITIALIZED })` inside of `channel.initialize()`